docs/new-project-setup.md

```diff
@@ -0,0 +1,305 @@
+GRANT ROLE orgadmin TO USER <YOUR-USER>;
+```
+
+If you later want to revoke the `orgadmin` role from your user or any other, you can do so with:
+
+```sql
+USE ROLE accountadmin;
+REVOKE ROLE orgadmin FROM USER <YOUR-USER>;
+```
+
+### Get access to AWS
+
+We typically create our Snowflake architecture using Terraform.
+Terraform state is stored in S3 buckets within our AWS account,
+so you will need read/write access to those buckets.
+
+Ask a DSE AWS admin to give you access to these buckets,
+and [configure your AWS credentials](./setup.md#aws).
+
+
+### Install terraform dependencies
+
+You can install Terraform using whatever approach makes sense for your system,
+including using `brew` or `conda`.
+
+Here is a sample for installing the dependencies using `conda`:
+
+```bash
+conda create -n infra python=3.10  # create an environment named 'infra'
+conda activate infra  # activate the environment
+conda install -c conda-forge terraform tflint  # Install terraform and tflint
+```
+
+## Snowflake account setup
+
+### Create the account
+
+1. Assume the `ORGADMIN` role
+1. Under the "Admin" side panel, go to "Accounts" and click the "+ Account" button:
+    1. Select the cloud and region appropriate to the project. The region should be in the United States.
+    1. Select "Business Critical" for the Snowflake Edition.
+    1. You will be prompted to create an initial user with `ACCOUNTADMIN` privileges. This should be you.
+        You will be prompted to create a password for your user. Create one using your password manager,
+        but know that it will ask you change your password upon first log-in.
+    1. Save the Account Locator and URL for your new account.
+1. Log into your new account. You should be prompted to change your password. Save the updated password in your password manager.
+
+### Enable multi-factor authentication for your user
+
+1. Ensure the Duo Mobile app is installed on your phone.
+1. In the upper-left corner, click on your user, and select "Profile"
+1. At the bottom of the dialog, select "Enroll" to enable multi-factor authentication.
+1. Follow the instructions to link the new account with your Duo app.
+
+### Set up key pair authentication
+
+Certain Snowflake clients don't properly cache MFA tokens,
+which means that using them can generate dozens or hundreds of MFA requests on your phone.
+At best this makes the tools unusable, and at worst it can lock your Snowflake account.
+One example of such a tool is (as of this writing) the Snowflake Terraform Provider.
+
+The recommended workaround for this is to add a key pair to your account for use with those tools.
+
+1. Follow the instructions given [here](https://docs.snowflake.com/en/user-guide/key-pair-auth#configuring-key-pair-authentication)
+    to generate a key pair and add the public key to your account.
+    Keep the key pair in a secure place on your device.
+    [This gist](https://gist.github.com/ian-r-rose/1c714ee04be53f7a3fd80322e1a22c27)
```

Comment:
I did not use this gist or the bash code below. if they are optional I think separating them out and denoting that would be helpful

Reply:
The gist is just a copy/paste of the bash commands in the linked docs. It's quite helpful when you are creating a lot of key pairs, which is the case for setting up an account. It is optional (which is what I meant to indicate by "can be helpful"), I can try to state it a bit more strongly.